### DISCUSSION TITLE
Inject/provides can you just consume or can you also provide?

### DISCUSSION THREAD

**Discussion by sgpinkus** [vuejs/router#1845] [Help and Questions] [answered]
[Docs state](https://router.vuejs.org/guide/advanced/navigation-guards.html#global-injections-within-guards)

> Since Vue 3.3, it is possible to use inject() within navigation guards. This is useful for injecting global properties like the [pinia stores](https://pinia.vuejs.org/).

Doesn't seem very useful to inject into routes, since your router is always pretty much right near the top of dep graph - it can just pull pinia DB as import or whatever. Much more useful would be the ability to `provide()` from router route. Can you do this, or is it just `inject()`

**Comment by posva** [comment, marked as answer]
You can only inject. It allows to inject any globals like i18n or anything else that is provided with `app.provide()`.

Note that importing from the root works in SPAs but doesn't on SSR (exposes to cross state pollution). That might be why you don't find this useful 😄

**Comment by sgpinkus** [comment]
> That might be why you don't find this useful

Didn't even think about that. Didn't really think past not wanting to inject (`provide()`) route specific values in global namespace.
